### PR TITLE
add additional examples

### PR DESCRIPTION
**Pull request by andre-merzky**
I think this covers the examples listed in #52.  Happy to iterate on them and/or to add additional examples, but would like to get a quick feedback.

### CODE REVIEW

File: specification.md [hunk 1]
```diff
@@ -1631,6 +1631,168 @@ ThrottleSubmitter().start()
...
+spec_2.executable = True   # type error
+
+job_1.specification = spec_1
+job_2.specification = spec_2
+
+try:
+    jex.submit(job_1)
+except jpsi.InvalidJobException:
+    # this should not happen
+    assert(True)
```
Comment (SteVwonder) [inline specification.md:1658]: Should this `assert(False)` to throw an assertion error in the exceptional case that this does occur?

File: specification.md [hunk 2]
```diff
@@ -1631,6 +1631,168 @@ ThrottleSubmitter().start()
...
+except jpsi.SubmitException:
+    # this *can* happen, dependent on backend state and policies
+    print('could not submit job - try again later')
+
+try:
+    jex.submit(job_2)
+except jpsi.InvalidJobException as e:
+    print('submission failed: %s' % e)
+else:
+    assert(True)  # the above should have raised an `InvalidJobException`
```
Comment (SteVwonder) [inline specification.md:1668]: Same question as above

File: specification.md [hunk 3]
```diff
@@ -1631,6 +1631,168 @@ ThrottleSubmitter().start()
...
+job.specification = spec
+
+jex.submit(job)
+job.wait()
+```
+
+
+#### N exclusive nodes, each with P processes
+
+This example will place a job across 5 nodes with 3 ranks per node.
```
Comment (SteVwonder) [inline specification.md:1710]:
```suggestion
This example will place a job across 5 nodes with 2 ranks per node.
```

File: specification.md [hunk 4]
```diff
@@ -1631,6 +1631,168 @@ ThrottleSubmitter().start()
...
+C = 4
+G = 1
+
+res_spec = jpsi.ResourceSpecification({
+              'process_count'    : P,
+              'cores_per_process': C,
+              'gpus_per_process' : G,
+           })
+job_spec = jpsi.JobSpecification({
+              'executable': 'workload.py',
```
Comment (SteVwonder) [inline specification.md:1692]:
```suggestion
              'executable': 'echo',
```

Assuming `workload.py` isn't provided somewhere else, this example will error out if someone blindly copies and pastes it.  Maybe we want to sacrifice some "realism" to increase the probability that it "just works"?  Just a though.  Feel free to ignore.

File: specification.md [hunk 5]
```diff
@@ -1631,6 +1631,168 @@ ThrottleSubmitter().start()
...
+              # other jobs can run on unused resources of the job's nodes
+              'exclusive_nodes': false,
+              # run a total of 10 ranks
+              'process_count': 10,
+              # place 3 ranks per node (last rank will be placed alone)
+              'processes__per_node': 3,
+              # each rank obtains 4 cores
+              'cores_per_process': 4,
+              # … and 2 GPUs
+              'gpus_per_process': 4,
```
Comment (SteVwonder) [inline specification.md:1757]:
```suggestion
              'gpus_per_process': 2,
```

**Review by SteVwonder — COMMENTED** [review]
Awesome! I think these are great. Just a few nits below:

File: specification.md (hunk 1 above)
Comment (andre-merzky) [reply specification.md:1658]: of course!

**Comment by andre-merzky** [comment]
Thanks @SteVwonder!

File: specification.md (hunk 2 above)
Comment (andre-merzky) [reply specification.md]: What was I thinking?    Nothing apparently :-P  Thanks.

File: specification.md [hunk 6]
```diff
@@ -1631,6 +1631,168 @@ ThrottleSubmitter().start()
...
+job_2.specification = spec_2
+
+try:
+    jex.submit(job_1)
+except jpsi.InvalidJobException:
+    # this should not happen
+    assert(False)
+except jpsi.SubmitException:
+    # this *can* happen, dependent on backend state and policies
+    print('could not submit job - try again later')
```
Comment (hategan) [inline specification.md:1661]: This would be a good place to use the `message` field of `SubmitException`.

File: specification.md [hunk 7]
```diff
@@ -1631,6 +1631,168 @@ ThrottleSubmitter().start()
...
+           })
+job_spec = jpsi.JobSpecification({
+              'executable': 'echo',
+              'arguments' : ['foo', 'bar', 'buz']
+              'directory' : '/tmp/',
+              'stdin'     : '/dev/null',
+              'stdout'    : 'work.out',
+              'stderr'    : 'work.err',
+              'resources' : res_spec,
+           })
```
Comment (hategan) [inline specification.md:1699]: This method of initializing the objects is not in the API, so I am not sure what message we want to send here:
- that frameworks have quite a few liberties here at the expense of source-level compatibility
- or that this IS in fact what the API is saying should be how JPSI objects are initialized

**Review by SteVwonder — APPROVED** [review]
LGTM!

**Review by hategan — APPROVED** [review]

**Comment by hategan** [comment]
OK, I don't want to hold this. When/if we decide on a better way to initialize the job info, we can update the examples as needed.

**Pull request merged**